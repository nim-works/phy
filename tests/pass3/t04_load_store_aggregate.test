discard """
  output: "(Error ecIllegalAccess)"
"""
(TypeDefs
  (Int 8)
  (Array 8 1 (Type 0))
  (Record 8 8
    (Field 0 (Type 0)))
  (Union  8 8 (Type 0))
  (Record 8 8
    (Field  0 (Type 1))
    (Field  8 (Type 2))
    (Field 16 (Type 3)))
  (ProcTy (Void)))
(GlobalDefs)
(ProcDefs
  (ProcDef (Type 5) 0
    (List
      (Block (Params) (Locals)
        (Store (Type 1)
          (IntVal 0)
          (Load (Type 1) (IntVal 0)))
        (Store (Type 2)
          (IntVal 0)
          (Load (Type 2) (IntVal 0)))
        (Store (Type 3)
          (IntVal 0)
          (Load (Type 3) (IntVal 0)))
        (Return)))))
