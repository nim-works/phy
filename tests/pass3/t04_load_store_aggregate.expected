;$sexp
(TypeDefs
  (Int 8)
  (ProcTy (Void))
  (UInt 8)
)
(GlobalDefs)
(ProcDefs
  (ProcDef (Type 1) 0
    (List
      (Block (Params) (Locals)
        (Copy (IntVal 0) (IntVal 0) (IntVal 8))
        (Copy (IntVal 0) (IntVal 0) (IntVal 8))
        (Copy (IntVal 0) (IntVal 0) (IntVal 8))
        (Return)))))
